### PR TITLE
Add support for multiple operators via several improvements

### PR DESCRIPTION
**Pull request by lisa**
This PR makes the following changes:

# scripts/gen_operator_csv.py (8fd92c1c88adc2cae821422e2c669554f658641e)

* Fix typo, add guardrails around missing assets
* If the `manifests` directory is missing, complain and quit. If there are no `Deployments` detected, complain and quit.

# Makefile (4d99ecdc6fa2fc7b642363e4a9dddfdc04fc8e8e)

* In all places substitute filesystem glob matching in `$(SOURCE_DIR)/**/` to iterate over `$(OPERATORS)` and determine the name of the repository from the list.
* Adds guardrails around certain variables that become part of the running environment due to `eval $(make -C operatordir) env`:
  * Tests are made to ensure the variables the make target cares about are found
  * Unsets variables so that subsequent loop iterations (and other make targets) aren't polluted with previous values
* Makes more explicit the testing around failures: Instead of `|| (... && exit)`, check the previous command's exit value in an `if-then-else`-style control block.
* Created an `env` target, which echos nothing, so that `functions.mk`'s `/process_template` doesn't emit an error.
* Most targets are now silent; they do not print the Bash that make is running. (To be more verbose try `make SHELL="bash -x"`)
* Consolidated multiple `mkdir -p manifests` into its own make target, `manifestdir`, with accompanying variable `MANIFESTDIR`

### CODE REVIEW

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
OPERATORS := openshift/dedicated-admin-operator openshift/configure-alertmanager-operator
```

Comment (jewzaam) [reply Makefile]: If we don't have this in the openshift org it needs created.  This is a requirement of the operator work. FYI @dak1n1

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
manifests: manifestdir manifests/catalog manifests/operators
```

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
		if [[ -z "$${OPERATOR_NAME}" || -z "$${OPERATOR_NAMESPACE}" || -z "$${OPERATOR_VERSION}" || -z "$${OPERATOR_IMAGE_URI}" ]]; then \
```

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
			echo "Couldn't determine OPERATOR_NAME, OPERATOR_NAMESPACE, OPERATOR_VERSION or OPERATOR_IMAGE_URI from $(SOURCE_DIR)/$$operator. make env output follows" ; \
```

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
			./scripts/gen_operator_csv.py $(SOURCE_DIR)/$$operator $$OPERATOR_NAME $$OPERATOR_NAMESPACE $$OPERATOR_VERSION $$OPERATOR_IMAGE_URI $(CHANNEL) 1>/dev/null ;\
```

File: Makefile
Comment (jewzaam) [inline Makefile]: Not needed if about to `exit 3`

File: Makefile
Comment (jewzaam) [inline Makefile]: Lots of unset.  Suggest moving this to the beginning of each loop iteration.  It only needs done when initializing this loop.  If a cleanup is desired, one outside the loop execution is fine too.

File: Makefile
Comment (jewzaam) [inline Makefile]: See comments and suggested changes from the `catalog` target.

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: Makefile
Comment (lisa) [reply Makefile]: Whoops, you're right. The repo exists, but she's still curating her PR.

File: Makefile
Comment (lisa) [reply Makefile]: I know, but for completeness sake 🤷‍♀️

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
		if [[ -z "$${OPERATOR_NAME}" || -z "$${OPERATOR_NAMESPACE}" || -z "$${OPERATOR_VERSION}" || -z "$${OPERATOR_IMAGE_URI}" ]]; then \
```

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
			echo "Couldn't determine OPERATOR_NAME, OPERATOR_NAMESPACE, OPERATOR_VERSION or OPERATOR_IMAGE_URI from $(SOURCE_DIR)/$$operator. make env output follows" ; \
```

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
			docker pull $$OPERATOR_IMAGE_URI ;\
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Comment by jewzaam** [comment]
PR into this PR to update readme: https://github.com/lisa/osd-operators-registry/pull/1

**Comment by jewzaam** [comment]
Changes in this repo are good, so:
/approve

But the config-alertmanager-operator repo has problems, noting here:

- inconsistent make target for build
- missing make target for push
- manifest filenames not catalogsource friendly (format is <text>.<kind.lowerCase>.yaml)

Bottom line is merging this breaks generation of the catalog.  Need the operator to be updated before we can merge. So:
/hold

**Comment by jewzaam** [comment]
Filenames can be handled here.  I'll put in a quick PR for that.  See https://github.com/openshift/osd-operators-registry/issues/15

**Comment by jewzaam** [comment]
Filename renames here: https://github.com/openshift/osd-operators-registry/pull/16

**Comment by lisa** [comment]
This will conflict with #17.

File: Makefile
Comment (jewzaam) [inline Makefile]:
```suggestion
				echo "Image cannot be pulled: $$OPERATOR_IMAGE_URI ;\
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Review by rogbas — APPROVED** [review]

**Pull request merged**
